framework/src/variables/MooseVariableFV.C

```diff
@@ -674,44 +629,23 @@ MooseVariableFV<OutputType>::getExtrapolatedBoundaryFaceValue(const FaceInfo & f
   mooseAssert(isExtrapolatedBoundaryFace(fi).first,
               "This function should only be called on extrapolated boundary faces");
 
-  auto it = _face_to_value.find(&fi);
-  if (it != _face_to_value.end())
-    return it->second;
-
+  ADReal boundary_value;
   const auto & tup = Moose::FV::determineElemOneAndTwo(fi, *this);
   const Elem * const elem = std::get<0>(tup);
-
   if (_two_term_boundary_expansion)
   {
-    // We need to compute the gradient. That gradient computation will cache the boundary face value
-    // for us
-    adGradSln(elem);
-
-    it = _face_to_value.find(&fi);
-    mooseAssert(it != _face_to_value.end(),
-                "adGradSln(elem) should have generated the boundary face value for us");
-
-    return it->second;
+    const Point vector_to_face = std::get<2>(tup) ? (fi.faceCentroid() - fi.elemCentroid())
+                                                  : (fi.faceCentroid() - fi.neighborCentroid());
+    boundary_value = adGradSln(elem) * vector_to_face + getElemValue(elem);
```

Comment:
I think you can reproduce the resilient behavior by returning 0 for the gradient when the two term fails

Reply:
I don't think the gradient is 0 in general if the 2term fails. There might be special cases when that is true but definitely not in general. And we still need/cache the gradient.

Reply:
yeah you re right. ok good on this.